qwerty_ast/src/ast.rs

```diff
@@ -349,6 +349,74 @@ impl<E> FunctionDef<E> {
     }
 }
 
+// NOTE: Until there is a nice way to distinguish these, there
+// are two display implementations. One could use Trivializable,
+// but you would have to modify it to return Option<Expr> and
+// this seemed antithetical to the purpose. I also didn't want
+// to create another trait just for this.
```

Comment:
I think we can mostly merge these two implementations now (see my comment below).